src/control/ros/RosBarrettHandPositionCommandExecutor.cpp

```diff
@@ -0,0 +1,181 @@
+#include <aikido/control/ros/RosBarrettHandPositionCommandExecutor.hpp>
+#include <aikido/control/ros/Conversions.hpp>
+
+namespace aikido {
+namespace control {
+namespace ros {
+
+using std::chrono::milliseconds;
+
+//=============================================================================
+template <typename DurationA, typename DurationB>
+RosBarrettHandPositionCommandExecutor::RosBarrettHandPositionCommandExecutor(
+      ::ros::NodeHandle node,
+      const std::string& serverName,
+      const DurationA& connectionTimeout,
+      const DurationB& connectionPollingPeriod)
+  : mNode(std::move(node))
+  , mCallbackQueue{}
+  , mClient{mNode, serverName, &mCallbackQueue}
+  , mConnectionTimeout{
+    std::chrono::duration_cast<milliseconds>(connectionTimeout)}
+  , mConnectionPollingPeriod{
+    std::chrono::duration_cast<milliseconds>(connectionPollingPeriod)}
```

Comment:
See my comment above about this cast. I'd prefer to eliminate the template parameters entirely and force _the user_ to perform this conversion, if necessary.

If we do decide to keep the template parameters, then this implementation needs to be moved into a header file. By convention, it would go into a file named `src/control/ros/detail/RosBarrettHandPositionCommandExecutor-impl.cpp`.